### ISSUE TITLE
feat: Maintain path after update has installed

### ISSUE THREAD

**Issue by alexcroox** [Cap-go/capacitor-updater#193]
### Description

With MS App Center's Codepush, once an update was installed, the current page/route would remain. This means when the user returns back to bring the app into focus, they won't realise anything happened.

Currently Capgo goes to the root (/) which for me is a login screen and I'd rather the user could continue where they left off.

### Platform(s)

ios/android

### Alternatives

Save current path before calling `CapacitorUpdater.set()`, redirect to the path after install

**Comment by riderx** [comment]
that a very good one, i have look but i don't know how to do it for now :)

**Comment by alexcroox** [comment]
I wonder how Codepush does it? Do they maybe grab `window.location.pathname` before the install, and send the webview there after install?

**Comment by riderx** [comment]
I don't know also yesterday looking into that i was wondering how to handle if you removed the path  the user was in

**Comment by alexcroox** [comment]
Yeah it’s a good point. Definitely a config option you’d want to leave up to user so they can think to handle that in the future if they enable it

**Comment by alexcroox** [comment]
Adding note from our previous discussion on Discord. Perhaps we could allow the plugin to persist some state (a JS object) when setting the new version, which is then retrievable via new function or .current()?

e.g `CapacitorUpdater.set({ id: data.id, state: { exampleRedirectMeToPathAfterInstallKey: '/platform/dashboard' })`

The challenge will be deciding where to retrieve this state, and how to delete it so old state isn't accessed in future updates if state isn't set next time (could it be versioned?)

**Comment by WcaleNieWolny** [comment]
Hello, Martin has requested that I take a look at this issue to determine if it is easily implementable.

**Comment by WcaleNieWolny** [comment]
I have just implemented this feature on Android.

https://github.com/user-attachments/assets/e295a5b9-318b-465c-9c65-5c6603f2808e

**Comment by alexcroox** [comment]
@WcaleNieWolny thank you!!

**Comment by riderx** [comment]
it's merged you can try it in 6.8.0
